modules/otel-collector-components.adoc

```diff
@@ -869,7 +902,47 @@ This extension supports traces, metrics, and logs.
 <10> The path to the client's private key file that is used with the client certificate if needed for authentication.
 <11> Sets a timeout for the token client's request.
 <12> You can assign the authenticator configuration to an OTLP exporter.
+[id="filestorage-extension_{context}"]
+=== Filestorage extension
+
+The File Storage extension can persist state to the local file system.
+The extension requires read and write access to a directory. A default directory can be used, but it must already exist in order for the extension to operate.
+
+The OAuth2Client extension persists the sending queue for the OTLP exporters that are based on the HTTP and the gRPC protocol.
+This extension supports traces, metrics, and logs.
 
+.OpenTelemetry Collector custom resource with configured FileStorage extension to persist as OTLP sendig queue.
+[source,yaml]
+----
+  config: |
+    extensions:
+      file_storage/all_settings:
+        directory: /var/lib/otelcol/mydir # <1>
+        timeout: 1s # <2>
+        compaction:
+          on_start: true # <3>
+          directory: /tmp/ # <4>
+          max_transaction_size: 65_536 # <5>
+        fsync: false # <6>
+
+    exporters:
+      otlp:
+        sending_queue:
+          storage: file_storage/all_settings # <7>
+
+    service:
+      extensions: [file_storage/all_settings]
+      pipelines:
+        traces:
+          receivers: [otlp]
+          exporters: [otlp]
+----
+<1> Specifies the directory in which the telemetry data is stored.
+<2> Specifies the a timeout to open the storedfiles.
+<3> Starts the compaction when the collector starts. If omitted, the default `+false+` is used.
+<4> Specifies the directory in which compactor stores the telemetry data is stored.
+<5> Defines maximum size of the compaction transaction. A value of zero will ignore transaction sizes. If omitted, the default `+65536+` is used.
```

Comment:
@frzifus, I think we will need to mention the units for this field. Can you find out?

Reply:
The unit is bytes. In [Compact](https://github.com/etcd-io/bbolt/blob/51926613987d57aa9df1f1bfcce0467598932e90/compact.go#L8-L50) they have a check if the addded keys+values exceed the max_transaction_size. If its the case, they commit the current transaction and create a new one.